cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -307,6 +307,90 @@ void generate_depth_remappings(std::map<int, std::pair<std::vector<int>, std::ve
   return total_pages;
 }
 
+/**
+ * @brief Count the total number of pages using page index information.
+ */
+[[nodiscard]] size_t count_page_headers_with_pgidx(host_span<ColumnChunkDesc> chunks)
+{
+  size_t total_pages = 0;
+  for (auto& chunk : chunks) {
+    CUDF_EXPECTS(chunk.h_chunk_info != nullptr, "Expected non-null column info struct");
+    auto const& chunk_info = *chunk.h_chunk_info;
+    chunk.num_dict_pages   = chunk_info.has_dictionary() ? 1 : 0;
+    chunk.num_data_pages   = chunk_info.pages.size();
+    total_pages += chunk.num_data_pages + chunk.num_dict_pages;
+  }
+  return total_pages;
+}
+
+// struct used to carry info from the page indexes to the device
+struct page_index_info {
+  int32_t num_rows;
+  int32_t chunk_row;
+  int32_t num_nulls;
+  int32_t num_valids;
+  int32_t str_bytes;
+};
+
+// functor to copy page_index_info into the PageInfo struct
+struct copy_page_info {
+  device_span<page_index_info> page_info;
+  device_span<PageInfo> pages;
```

Comment:
The naming is so tangled here. One member is named like the type of the other one. 
Maybe `index_infos` or `page_indexes` (given that `PageInfo` member is called `pages`)?